src/hooks/__internal__/useClickAwayListener/useClickAwayListener.ts

```diff
@@ -1,29 +1,20 @@
-import React, { useEffect, useRef } from "react";
+import React, { useEffect } from "react";
+
+const stopPropagation = (ev: React.MouseEvent) => {
+  ev.stopPropagation();
```

Comment:
Whilst this on the whole will work I don't think we should be stopping event propagation in Carbon (or at least we should be extremely careful when we do) as we don't know what the UI composition for consumers will be and therefore what we could be affecting. I also believe this will require the user to double click to outside the the container to close it if they have an additional popover (Select for example) open when I would expect both to close with single click outside of the PopoverContainer boundary.

https://stackoverflow.com/a/54817255

Reply:
Yes, thanks - I guess I agree having thought about it and read that SO comment that it's ideally something to avoid. I don't see how else to do this in a way that works with Portals but will look at how MUI do it. If there isn't another good way I'm still inclined to use stopPropagation because I honestly don't see what event handlers for click or mousedown a consumer is going to want on a parent component. But point taken, I'm going to do some research and if there's a way to avoid stopping propagation - without writing hideously overcomplicated code which likely still won't cover all possible cases - I will.